src/components/icon/icon-provider.ts

```diff
@@ -0,0 +1,255 @@
+import {Injectable, Renderer} from 'angular2/core';
+import {Http, Response, HTTP_PROVIDERS} from 'angular2/http';
+import {AsyncSubject, Observer, Observable} from 'rxjs/Rx';
+
+/**
+  *  Configuration for an icon, possibly including the cached SVG element.
+  */
+class IconConfig {
+  svgElement: SVGElement = null;
+  constructor(public url: string, public viewBoxSize: number) {
```

Comment:
Yeah it seems like an obscure feature. You can use it to have a version of the icon that's zoomed in or out, but only from one corner because the top and left coordinates are always 0. I don't know of any especially compelling use cases, happy to remove it.